### PR TITLE
fix: esm package traversal

### LINKED ISSUE #306: ESM `package.json` in home directory crashes LSP
I was looking into a [Discord bug I found](https://x.com/Hacksore/status/1790395583611474423) and funnily enough it also breaks the copilot LSP 😂.

```
[START][2024-08-03 11:32:06] LSP logging initiated
[ERROR][2024-08-03 11:32:06] .../vim/lsp/rpc.lua:770	"rpc"	"node"	"stderr"	"file:///Users/hacksore/.local/share/nvim/lazy/copilot.lua/copilot/index.js:3
global.__rootDirectory = __dirname + '/dist/';
                         ^

ReferenceError: __dirname is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and '/Users/hacksore/package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at file:///Users/hacksore/.local/share/nvim/lazy/copilot.lua/copilot/index.js:3:26
    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:323:24)
    at async loadESM (node:internal/process/esm_loader:28:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)

Node.js v20.12.2
```

Not sure this is fixable but raising the issue for awareness just in case others run into this.

**tl;dr** don't have a `package.json` in your homedir with `"type": "module"` 💀.

### PR DESCRIPTION
**Pull request by Hacksore**
Closes #306

I tested this by:
1. creating a `package.json` in my home dir
2. adding `"type": "module"`
3. opening nvim and see copilot crash
4. adding the `package.json` see in the diff to `~/.local/share/nvim/lazy/copilot.lua/copilot/package.json`
5. open nvim
6. observed no crashes 🥳

### CODE REVIEW

**Comment by Hacksore** [comment]
also linking this issue from node for others trying to fix this in their apps.
https://github.com/nodejs/node/issues/54632

**Comment by Hacksore** [comment]
can any devs in the chat take a look when they get some free time?

@MunifTanjim @zbirenbaum @mvaldes14

**Comment by zbirenbaum** [comment]
Thanks for researching this! LGTM
